cdap-ui/app/directives/explore/explore.js

```diff
@@ -0,0 +1,155 @@
+angular.module(PKG.name + '.commons')
+  .directive('myExplore', function () {
+
+    return {
+      restrict: 'E',
+      scope: {
+        type: '=',
+        name: '='
+      },
+      templateUrl: 'explore/explore.html',
+      controller: myExploreCtrl,
+      controllerAs: 'MyExploreCtrl'
+    };
+
+
+    function myExploreCtrl ($scope, MyDataSource, myExploreApi, $http, $state, $bootstrapModal, myCdapUrl, $timeout, myAlert) {
+        var vm = this;
+
+        var dataSrc = new MyDataSource($scope);
+        var isFirstTime = true;
+        vm.queries = [];
+        vm.currentPage = 1;
+        var params = {
+          namespace: $state.params.namespace,
+          scope: $scope
+        };
+
+        $scope.$watch('name', function() {
+          vm.query = 'SELECT * FROM ' + $scope.type + '_' + $scope.name + ' LIMIT 5';
+        });
+
+        vm.execute = function() {
+          myExploreApi.postQuery(params, { query: vm.query }, vm.getQueries);
+          vm.currentPage = 1;
+        };
+
+        vm.getQueries = function() {
+
+          myExploreApi.getQueries(params)
+            .$promise
+            .then(function (queries) {
+              if (!isFirstTime) {
```

Comment:
The if block will prevent all the queries being green the first time user loads the page.